### PR TITLE
Run examples under ASAN and TSAN

### PR DESCRIPTION
**Pull request by daviddrysdale**
Under development, just putting as a PR to get a cloudbuild running.

### CODE REVIEW

File: scripts/build_server_dev [hunk 1]
```diff
@@ -4,10 +4,32 @@ readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
...
 source "$SCRIPTS_DIR/common"
 
+CONFIG='clang'
+
+while getopts "at" opt; do
+    case "$opt" in
+    a)
+        CONFIG="asan"
+        bazel_build_flags+=(
+          "--compilation_mode=dbg"
```
Comment (anghelcovici) [inline scripts/build_server_dev]: should we add this as part of the configs instead? if you specify asan you should build dbg by default

Comment (daviddrysdale) [reply scripts/build_server_dev]: Done.

**Comment by daviddrysdale** [comment]
Plan is to wait on #359 arriving then drop the "temp: &hellip;" commit.

**Comment by tiziano88** [comment]
FYI https://github.com/google/sanitizers/issues/1039#issuecomment-453607726: it should be possible to run asan+lsan+ubsan together, and then msan separately.

**Comment by tiziano88** [comment]
ref #357 tracks the actual fix for this so that cache is automatically populated after merging a commit

**Pull request merged**